Sources/ProjectSpec/SpecFile.swift

```diff
@@ -63,26 +79,28 @@ public struct SpecFile {
     private init(include: Include, basePath: Path, relativePath: Path, cachedSpecFiles: inout [Path: SpecFile], variables: [String: String]) throws {
         let basePath = include.relativePaths ? (basePath + relativePath) : basePath
         let relativePath = include.relativePaths ? include.path.parent() : Path()
-        let includePath = include.relativePaths ? basePath + relativePath + include.path.lastComponent : basePath + include.path
 
-        try self.init(filePath: includePath, basePath: basePath, cachedSpecFiles: &cachedSpecFiles, variables: variables, relativePath: relativePath)
+        try self.init(filePath: include.path, basePath: basePath, cachedSpecFiles: &cachedSpecFiles, variables: variables, relativePath: relativePath)
     }
 
-    public init(filePath: Path, basePath: Path, cachedSpecFiles: inout [Path: SpecFile], variables: [String: String] = [:], relativePath: Path = "") throws {
-        let jsonDictionary = try SpecFile.loadDictionary(path: filePath).expand(variables: variables)
+    private init(filePath: Path, basePath: Path, cachedSpecFiles: inout [Path: SpecFile], variables: [String: String], relativePath: Path = "") throws {
+        let path = basePath + filePath
+        if let specFile = cachedSpecFiles[path] {
+            self = specFile
+            return
+        }
+
+        let jsonDictionary = try SpecFile.loadDictionary(path: path).expand(variables: variables)
+
         let includes = Include.parse(json: jsonDictionary["include"])
         let subSpecs: [SpecFile] = try includes
             .filter(\.enable)
             .map { include in
-                if let specFile = cachedSpecFiles[filePath] {
-                    return specFile
-                } else {
-                    return try SpecFile(include: include, basePath: basePath, relativePath: relativePath, cachedSpecFiles: &cachedSpecFiles, variables: variables)
-                }
```

Comment:
`filePath`, which is for the current spec, is not the correct key to use when looking for the `specFile` of an `include` that has it's own path.